### PR TITLE
fix verdi computer show tab completion

### PR DESCRIPTION
**Pull request by unkcpz**
Got the following error when doing `verdi computer show <tab>`. This is a quick fix but just thinking when this change made in `DbComputer`, maybe other places also need to be take care of?

```
Traceback (most recent call last):                                                                                      
  File "/home/unkcpz/Projects/python-code/aiida_core/aiida/orm/implementation/querybuilder.py", line 399, in get_column 
    return getattr(alias, colname)                                                                                      
  File "/home/unkcpz/miniconda3/envs/aiida-core-dev/lib/python3.8/site-packages/sqlalchemy/orm/util.py", line 506, in __
getattr__                                                                                                               
    attr = getattr(target, key)                                                                                         
AttributeError: type object 'DbComputer' has no attribute 'name'
...
...
  File "/home/unkcpz/Projects/python-code/aiida_core/aiida/orm/querybuilder.py", line 1225, in _add_to_projections
    entity_to_project = self._get_projectable_entity(alias, column_name, attr_key, cast=cast)
  File "/home/unkcpz/Projects/python-code/aiida_core/aiida/orm/querybuilder.py", line 1199, in _get_projectable_entity
    entity = self._impl.get_column(column_name, alias)
  File "/home/unkcpz/Projects/python-code/aiida_core/aiida/orm/implementation/querybuilder.py", line 401, in get_column
    raise ValueError(
ValueError: name is not a column of aliased(DbComputer)
Valid columns are:
id
uuid
label
hostname
description
scheduler_type
transport_type
metadata
```

### CODE REVIEW

File: aiida/cmdline/params/types/computer.py [hunk 1]
```diff
@@ -41,7 +41,7 @@ def complete(self, ctx, incomplete):  # pylint: disable=unused-argument
 
         :returns: list of tuples of valid entry points (matching incomplete) and a description
         """
-        return [(option, '') for option, in self.orm_class_loader.get_options(incomplete, project='name')]
+        return [(option, '') for option, in self.orm_class_loader.get_options(incomplete, project='hostname')]
```
Comment (sphuber) [inline aiida/cmdline/params/types/computer.py:44]:
```suggestion
        return [(option, '') for option, in self.orm_class_loader.get_options(incomplete, project='label')]
```

**Review by sphuber — CHANGES_REQUESTED** [review]
Thanks @unkcpz . The change of `DbComputer.name` to `DbComputer.label` was done in #4882 . It should have updated all other places in the code that still used `name` but this one was clearly missed and it is not tested. It would be great if you could add a quick test for it. Note that I think therefore we should also project on `label` instead of the hostname, because that is essentially what was being projected originally.

**Comment by unkcpz** [comment]
Thanks @sphuber. The test added and also refactoring the unit test to use pytest.

**Comment by unkcpz** [comment]
Well, seems run into pre-commit pylint warnings. Do we need to exclude `tests/` in pre-commit-pylint?

File: tests/cmdline/params/types/test_computer.py [hunk 2]
```diff
@@ -8,88 +8,110 @@
 # For further information please visit http://www.aiida.net               #
 ###########################################################################
 """Tests for the `ComputerParamType`."""
```
Comment (sphuber) [inline tests/cmdline/params/types/test_computer.py:10]: # pylint: disable=redefined-outer-name

File: tests/cmdline/params/types/test_computer.py [hunk 3]
```diff
@@ -8,88 +8,110 @@
...
-        result = self.param.convert(identifier, None, None)
-        self.assertEqual(result.uuid, self.entity_03.uuid)
+@pytest.fixture
+def parameter_type():
+    """Return an instance of the `ComputerParamType`."""
+    return ComputerParamType()
+
+
+@pytest.fixture
+def setup_computers(clear_database_before_test):
```
Comment (sphuber) [inline tests/cmdline/params/types/test_computer.py:26]:
```suggestion
@pytest.mark.usefixtures('clear_database_before_test')
def setup_computers():
```

File: tests/cmdline/params/types/test_computer.py [hunk 4]
```diff
@@ -8,88 +8,110 @@
...
+
+    options = [item[0] for item in parameter_type.complete(None, 'xa')]
+    assert sorted(options) == sorted([entity_04.label])
+
+
+def test_get_by_id(setup_computers, parameter_type):
+    """
+    Verify that using the ID will retrieve the correct entity
+    """
+    entity_01, entity_02, entity_03 = setup_computers
```
Comment (sphuber) [inline tests/cmdline/params/types/test_computer.py:60]:
```suggestion
    entity_01, _, _ = setup_computers
```

**Review by sphuber — CHANGES_REQUESTED** [review]
Thanks a lot for the test and refactoring @unkcpz  . The tests should also be linted so it should not be added to the exclude list. The errors are easy to address though. I made three suggestions that show how to do it. Just the one of unused variables you should apply to the other locations in the code that pylint mentioned

**Comment by unkcpz** [comment]
@sphuber thanks!  It should be all right this time.

File: tests/cmdline/params/types/test_computer.py (hunk 3 above)
Comment (sphuber) [reply tests/cmdline/params/types/test_computer.py]: Uhm, we do it in plenty of places with no problems. So I doubt that it is not possible. What makes you think that besides that PR you linked?

**Review by sphuber — APPROVED** [review]

**Pull request merged**